### PR TITLE
SLING-10652 add table for HTL type conversions

### PR DESCRIPTION
**Pull request by kwin**
Here is draft which may act as starting point....

### CODE REVIEW

**Review by raducotescu — APPROVED** [review]

**Comment by raducotescu** [comment]
@kwin, I've expanded a bit the table and mentioned how `Optional` is handled. WDYT?

File: src/main/jbake/content/documentation/bundles/scripting/scripting-htl.md [hunk 1]
```diff
@@ -56,6 +56,31 @@ The [HTML Template Language Specification](https://github.com/Adobe-Marketing-Cl
...
+`Date` | `java.util.Date`, `java.util.Calendar` |
+`Number` | `java.lang.Number`, every other type first converted to `java.lang.String` and then converted to Number via [`NumberUtils.createNumber(String)`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/math/NumberUtils.html#createNumber-java.lang.String-). | [`ObjectModel.toNumber(Object)`](https://github.com/apache/sling-org-apache-sling-scripting-sightly-runtime/blob/org.apache.sling.scripting.sightly.runtime-1.2.4-1.4.0/src/main/java/org/apache/sling/scripting/sightly/render/ObjectModel.java#L204)
+`Collection` | Empty list for everything except for <br/>arrays, Collections, Iterators and Iterables (converted to List) and `java.lang.String` or `java.lang.Number` (a list containing a single item). For `java.util.Map` the key set is used.  | [`ObjectModel.toCollection(Object)`](https://github.com/apache/sling-org-apache-sling-scripting-sightly-runtime/blob/org.apache.sling.scripting.sightly.runtime-1.2.4-1.4.0/src/main/java/org/apache/sling/scripting/sightly/render/ObjectModel.java#L277)
+`Map` |
```
Comment (kwin) [inline src/main/jbake/content/documentation/bundles/scripting/scripting-htl.md:70]: @raducotescu Can you add some info for `Map` and `Enum`?

File: src/main/jbake/content/documentation/bundles/scripting/scripting-htl.md [hunk 2]
```diff
@@ -56,6 +56,31 @@ The [HTML Template Language Specification](https://github.com/Adobe-Marketing-Cl
...
+`Date` | `java.util.Date`, `java.util.Calendar` |
+`Number` | `java.lang.Number`, every other type first converted to `java.lang.String` and then converted to Number via [`NumberUtils.createNumber(String)`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/math/NumberUtils.html#createNumber-java.lang.String-). | [`ObjectModel.toNumber(Object)`](https://github.com/apache/sling-org-apache-sling-scripting-sightly-runtime/blob/org.apache.sling.scripting.sightly.runtime-1.2.4-1.4.0/src/main/java/org/apache/sling/scripting/sightly/render/ObjectModel.java#L204)
+`Collection` | Empty list for everything except for <br/>arrays, Collections, Iterators and Iterables (converted to List) and `java.lang.String` or `java.lang.Number` (a list containing a single item). For `java.util.Map` the key set is used.  | [`ObjectModel.toCollection(Object)`](https://github.com/apache/sling-org-apache-sling-scripting-sightly-runtime/blob/org.apache.sling.scripting.sightly.runtime-1.2.4-1.4.0/src/main/java/org/apache/sling/scripting/sightly/render/ObjectModel.java#L277)
```
Comment (kwin) [inline src/main/jbake/content/documentation/bundles/scripting/scripting-htl.md:69]: why is Date and Boolean not converted to a single item list?

Comment (raducotescu) [reply src/main/jbake/content/documentation/bundles/scripting/scripting-htl.md:69]: No idea why we haven't implemented that conversion as well. :)

Comment (kwin) [reply src/main/jbake/content/documentation/bundles/scripting/scripting-htl.md:69]: Ok, created https://issues.apache.org/jira/browse/SLING-10679 for that

**Pull request merged**